cluster-autoscaler/cloudprovider/clusterapi/clusterapi_machinedeployment.go

```diff
@@ -101,6 +101,28 @@ func (r machineDeploymentScalableResource) SetSize(nreplicas int32) error {
 	return updateErr
 }
 
+func (r machineDeploymentScalableResource) UnmarkMachineForDeletion(machine *Machine) error {
```

Comment:
I'm assuming both machineDeploymentScalableResource and machineSetScalableResource implement an interface.

We could have better code reuse if we do

func UnmarkMachineForDeletion (r machineScalableInterface, machine *Machine)

Obviously, replace machineScalableInterface with whatever the actual interface is.

Reply:
they do have an interface, i'll see if i can just define this function there.

Reply:
This function should be part of the scalable resource interface, but this pattern does match the existing code within the scalable resources, eg see `Nodes()` which IIRC wraps a method in the clusterapi_scalableresource file